openedx/core/djangoapps/user_api/validation/views.py

```diff
@@ -0,0 +1,145 @@
+            >>>     }
+            >>> }
+            In this example, username is valid and (we assume) there is
+            a preexisting account with that email. The password also seems
+            to contain the username.
+
+            Note that a validation decision is returned *for all* inputs, whether
+            positive or negative.
+
+        **Available Handlers**
+
+            "username":
+                A handler to check the validity of usernames.
+            "email":
+                A handler to check the validity of emails.
+            "password":
+                A handler to check the validity of passwords; a compatibility
+                decision with the username is made if it exists in the input.
+    """
+
+    def username_handler(self, request):
+        username = request.GET.get('username')
+        invalid_username_error = get_username_validation_error(username)
+        username_exists_error = get_username_existence_validation_error(username)
+        # Existing usernames are already valid, so we prefer that error.
+        return username_exists_error or invalid_username_error
+
+    def email_handler(self, request):
+        email = request.GET.get('email')
+        invalid_email_error = get_email_validation_error(email)
+        email_exists_error = get_email_existence_validation_error(email)
+        # Existing emails are already valid, so we prefer that error.
+        return email_exists_error or invalid_email_error
+
+    def password_handler(self, request):
+        username = request.GET.get('username') or None
+        password = request.GET.get('password')
+        return get_password_validation_error(password, username)
+
+    validation_handlers = {
+        "username": username_handler,
+        "email": email_handler,
+        "password": password_handler,
+    }
+
+    def get(self, request):
+        """
+        GET /api/user/v1/validation/registration/
+
+        Expects request of the form
+        >>> {
+        >>>     "username": "mslm",
+        >>>     "email": "mslm@gmail.com",
+        >>>     "password": "password_mslm_in_it_thus_invalid"
+        >>> }
+        where each key is the appropriate form field name and the value is
+        user input and any of its validation dependencies (i.e. for password,
+        we depend on username). One may enter individual inputs if needed.
+        """
+        validation_decisions = {}
+        for form_field_key in self.validation_handlers:
+            # For every field requiring validation from the client,
+            # request a decision for it from the appropriate handler.
+            if form_field_key in request.GET:
+                handler = self.validation_handlers[form_field_key]
+                validation_decisions.update({
+                    form_field_key: handler(self, request)
+                })
+        return Response({"validation_decisions": validation_decisions})
```

Comment:
Yea, I also agree with this point. Though I'm wondering if that means each of these handlers would then become their own views, since it would be ambiguous to return 400 with a request that had one valid input and one invalid input.